keyring/cli.py

```diff
@@ -81,14 +109,31 @@ def run(self, argv):
 
     def _check_args(self):
         if self.operation:
-            if self.service is None or self.username is None:
+            if self.operation == 'get' and self.get_mode == 'creds':
+                if self.service is None:
+                    self.parser.error(f"{self.operation} requires service")
+            elif self.service is None or self.username is None:
                 self.parser.error(f"{self.operation} requires service and username")
```

Comment:
This 6 line function now has a cyclomatic complexity of 5, up from 3 in the prior implementation. I'm going to see if it's possible to keep it lower.

Reply:
Done in 33bae95.